dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -949,6 +1005,33 @@ BlockInputStreamPtr Segment::getInputStream(
         expected_block_size,
         columns_to_read,
         segment_snap->stable->stable);
+
+    // Building MVCC bitmap can be a resource-intensive operation that cannot be paused midway.
+    // To prevent scenarios where multiple segments concurrently build MVCC bitmaps but exhaust
+    // RU during execution and causing RU consumption to exceed limitations, we need to first
+    // estimate the cost of building the MVCC bitmap, pre-consume the corresponding RU,
+    // and then proceed with the building task.
+    const auto & res_group_name = dm_context.scan_context->resource_group_name;
+    if (likely(read_mode == ReadMode::Bitmap && !res_group_name.empty()))
+    {
+        auto real_ranges = shrinkRowKeyRanges(read_ranges);
+        if (!real_ranges.empty())
+        {
+            auto pack_filters = loadDMFilePackFilters(
```

Comment:
Just a note for the difference between cherry-pick and the changes in master: here the DMFilePackFilter is loaded one more time.